src/use_case/kill_villager/KillVillagerInteractor.java

```diff
@@ -1,2 +1,61 @@
-package use_case.kill_villager;public class KillVillagerInteractor {
-}
+package use_case.kill_villager;
+
+import entity.Game;
+import entity.PromptGenerator;
+import use_case.ChatAPIAccessInterface;
+
+public class KillVillagerInteractor implements KillVillagerInputBoundary{
+
+    private final KillVillagerDataAccessInterface killVillagerDataAccessInterface;
+    private final KillVillagerOutputBoundary killVillagerPresenter;
+    private final PromptGenerator promptGenerator;
+    private final ChatAPIAccessInterface gptDataAccessObject;
+    // This is named storyDataAccessObject in VotePlayer use case, should probably choose one name or the other for
+    // uniformity.
+    private final Game game;
+    final String villager;
+
+    public KillVillagerInteractor(KillVillagerDataAccessInterface killVillagerDataAccessInterface,
+                                  KillVillagerOutputBoundary killVillagerPresenter,
+                                  KillVillagerInputData killVillagerInputData,
+                                  ChatAPIAccessInterface gptDataAccessObject) {
+        this.killVillagerDataAccessInterface = killVillagerDataAccessInterface;
+        this.killVillagerPresenter = killVillagerPresenter;
+        // TODO: Figure out how to talk to the Output Boundary without violating CA, i.e. where the output boundary is
+        //  being passed in from.
+        // Notice how we are getting the prompt generator and game from the killVillagerDataAccessInterface,
+        // adhering to the CA Engine.
+        this.promptGenerator = killVillagerDataAccessInterface.getPromptGenerator();
+        this.game = killVillagerDataAccessInterface.getGame();
+        this.villager = killVillagerInputData.getVillager();
+        // Pulling Villager name here instead of in killVillager() so the code down there is less messy.
+        // TODO: Ensure input data implementation is consistent with VotePlayer, and either both use input data or both
+        //  do not.
+        this.gptDataAccessObject = gptDataAccessObject;
+        // TODO: Figure out how to talk to ChatGPT without violating CA, since passing in gptDAO as a parameter means it
+        //  is coming in as an input from the controller, which is not supposed to be able to talk to the DAO.
+    }
+
+    @Override
+    public void killVillager() {
```

Comment:
Okay slight mistake the input data shouldn't be going into the interactor's constructor but instead it should get passed into killVillager( -- input data here -- ). Because of that, String villager shouldn't be an attribute of the class either, it only exists inside the killVillager() method. So essentially, the player's name aka the input data goes from the view into the controller's execute function. The controller's execute function creates the input data from this player name and then it passes that input data into the interactor's execute function. If we had the player name as an attribute of the interactor then the controller would basically be useless. Take a look at my controller or maybe the Login use case in week5ca if you wanna see more specifics about why this is done.